cdapio/cdap-ui
# Review: CDAP 19917 - Snackbar

# TITLE
Feature - Snackbar Component

## Description
This PR contains the Snackbar Component used to show Success and Failure Messages.

## Files to be reviewed as part of this PR
Please review all files that are part of this PR

## Related PR
This pr was made because 
1. The previous one: https://github.com/cdapio/cdap-ui/pull/725 had too many conflicts in the squash merge. 
2. There were unwanted file changes as part of the previous PR and have been ignored in this PR 

This pr cleans the commit history and ignores the unwanted file changes

## PR Type

## Links
Jira: [Jira issue #](fill-in.org)
https://cdap.atlassian.net/browse/CDAP-19917

## Test Plan
1. Verify whether the Snack Bar is displayed or not, when u apply on any of the wrangle links from connector types. - Pass
5. Verify after applying on the wrangler link with valid data, a success message is displayed or not. - Pass
6. Verify whether the Success Message is displayed along with the "Dataset Loaded Successfully." text in the snack bar or not. - Pass
7. Verify after applying on the wrangler link with Invalid/No data, Warning/Failure message is displayed or not. - Pass
8. Verify whether the Failure message is displayed along with the file name we selected in the snack bar or not. - Pass
9. Verify the snack bar with the Warning/Failure message is displayed in Red color or not. - Pass
10. Verify the snack bar with Success Message is displayed in Green color or not. - Pass
11. Verify the info icon beside the Failure text, - Pass
12. Verify the tick icon beside the Success text. - Pass
13. Verify the "info & tick" icon by mouse hovering. - Pass
14. Verify the "close icon" is displayed on the right end of the snack bar - Pass
15. Verify by clicking on the 'Close icon' of SnackBar - Pass
16. Verify the action after clicking on the 'Close icon'. of SnackBar - Pass

## Screenshots

1. Success Snackbar
<img width="1440" alt="Screenshot 2022-11-04 at 6 00 51 PM" src="https://user-images.githubusercontent.com/61772422/199973440-419db683-877e-4e65-a10c-53795743ee34.png">

5. Failure Snackbar
<img width="1439" alt="Screenshot 2022-11-04 at 6 01 50 PM" src="https://user-images.githubusercontent.com/61772422/199973451-85282bba-368d-4932-a716-34b78a29b61c.png">
